### PR TITLE
Add new version of clipper.exchange on Ethereum (ClipperApproximateCa…

### PR DESCRIPTION
**Pull request by amalashkevich**
…ravelExchange)

Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by amalashkevich** [comment]
Hi @Hosuke, could you please help to understand what is wrong with the test here? 
`select * from test_schema.git_e7ac19ac_check_dex_seed_clipper_v4_ethereum_trades_ethereum__clipper__4`

File: models/clipper/ethereum/clipper_ethereum_trades.sql [hunk 1]
```diff
@@ -11,6 +11,7 @@
 'clipper_v1_ethereum_trades'
 ,'clipper_v2_ethereum_trades'
 ,'clipper_v3_ethereum_trades'
+,'clipper_v4_ethereum_trades'
```
Comment (Hosuke) [inline models/clipper/ethereum/clipper_ethereum_trades.sql:14]: Sorry I cannot commit directly, but try to replace this file with the following:
```
{{ config(
        alias ='trades',
        post_hook='{{ expose_spells(\'["ethereum"]\',
                                "project",
                                "clipper",
                                \'["0xRob", "amalashkevich"]\') }}'
        )
}}

{% set clipper_models = [
ref('clipper_v1_ethereum_trades')
, ref('clipper_v2_ethereum_trades')
, ref('clipper_v3_ethereum_trades')
, ref'clipper_v4_ethereum_trades')
] %}


SELECT *
FROM (
    {% for dex_model in clipper_models %}
    SELECT
        blockchain,
        project,
        version,
        block_date,
        block_time,
        token_bought_symbol,
        token_sold_symbol,
        token_pair,
        token_bought_amount,
        token_sold_amount,
        token_bought_amount_raw,
        token_sold_amount_raw,
        amount_usd,
        token_bought_address,
        token_sold_address,
        taker,
        maker,
        project_contract_address,
        tx_hash,
        tx_from,
        tx_to,
        trace_address,
        evt_index
    FROM {{ dex_model }}
    {% if not loop.last %}
    UNION ALL
    {% endif %}
    {% endfor %}
)
;
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @amalashkevich I try to suggest a possible fix to the problem you mentioned.

File: models/clipper/ethereum/clipper_ethereum_trades.sql [hunk 2]
```diff
@@ -3,14 +3,15 @@
...
 }}
 
 {% set clipper_models = [
-'clipper_v1_ethereum_trades'
-,'clipper_v2_ethereum_trades'
-,'clipper_v3_ethereum_trades'
+ref('clipper_v1_ethereum_trades')
+, ref('clipper_v2_ethereum_trades')
+, ref('clipper_v3_ethereum_trades')
+, ref'clipper_v4_ethereum_trades')
```
Comment (Hosuke) [inline models/clipper/ethereum/clipper_ethereum_trades.sql:14]: Sorry I missed a `(` here:
```suggestion
, ref('clipper_v4_ethereum_trades')
```

Comment (amalashkevich) [reply models/clipper/ethereum/clipper_ethereum_trades.sql]: yep

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @amalashkevich 

- Metadata & naming conventions ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercase ✅
- Inner join to large tables(ethereum.transactions) as possible ✅
- `CAST()` used in the base models ✅

[Sample clipper_v4_ethereum_trades](https://dune.com/queries/2338903?d=1)

**Comment by amalashkevich** [comment]
Thank you @Hosuke for your help!

**Comment by amalashkevich** [comment]
@Hosuke I've merged the branch and now getting an error, but it is not with my code. What should I do?

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**